app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt

```diff
@@ -65,8 +78,41 @@ class NewOnboardingFragment :
         signInHelper.showSignInDialog(requireContext(), googleSignInLauncher)
     }
 
+    override fun showError(error: String) {
+        view?.post {
+            setLoadingState(isScreenLoading = false)
+            showErrorSnackBar(error)
+        }
+    }
+
+    override fun onSameTokenError() {
+        view?.post {
+            setLoadingState(isScreenLoading = false)
+            replaceFragment(WalletFoundFragment.create())
+        }
+    }
+
+    override fun onSuccessfulSignUp() {
+        view?.post {
+            // TODO PWN-4268 move user to phone number screen
+            setLoadingState(isScreenLoading = false)
+            Toast.makeText(requireContext(), "You are successfully signed in!", Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    private fun setLoadingState(isScreenLoading: Boolean) {
+        with(binding) {
```

Comment:
> with(binding.buttonCreateWalletOnboarding) {
> isLoading = isScreenLoading
> isEnabled = !isScreenLoading
> }

Reply:
buttonRestoreWalletOnboarding.isEnabled = !isScreenLoading

you can remove this line

Reply:
why?